### PR TITLE
feat(bigquery): Add support for Avro options to external data sources

### PR DESCRIPTION
**Pull request by quartzmo**
* Add External::AvroSource

closes: #14158

### CODE REVIEW

File: google-cloud-bigquery/lib/google/cloud/bigquery/external/data_source.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ module External
         #
         #   bigquery = Google::Cloud::Bigquery.new
         #
-        #   avro_url = "gs://bucket/path/to/data.avro"
+        #   avro_url = "gs://bucket/path/to/*.avro"
```
Comment (dazuma) [inline google-cloud-bigquery/lib/google/cloud/bigquery/external/data_source.rb:40]: Why are we changing this? Does the call support wildcards like this?

**Review by dazuma — APPROVED** [review]
LGTM, with a question

File: google-cloud-bigquery/lib/google/cloud/bigquery/external/data_source.rb (hunk 1 above)
Comment (quartzmo) [reply google-cloud-bigquery/lib/google/cloud/bigquery/external/data_source.rb:40]: Yes, see [Wildcard support for Cloud Storage URIs](https://cloud.google.com/bigquery/external-data-cloud-storage#wildcard-support). And Avro data will typically be in multiple files, see examples in [Loading Avro data from Cloud Storage](https://cloud.google.com/bigquery/docs/loading-data-cloud-storage-avro). I believe showing a single file is less realistic.

**Review by shollyman — APPROVED** [review]
Wildcards, wildcards for everyone!  Thanks for putting this together.

**Pull request merged**